### PR TITLE
Servicestack

### PR DESCRIPTION
**Pull request by mmesson**
@aallfredo here is. I really dont know how to handle errors with servicestack to get errors like 404 or 401 and handle it well with nice alerts.

### CODE REVIEW

File: ChromeExtension/js/GeniusLinkServiceClient.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        objectToProcess.format = "jsv";
+
+        GeniusLinkServiceClient.prototype.getFromService.call(this, url, objectToProcess, callback, error);
+    }
+
+
+
+
+}
+GeniusLinkServiceClientPost.prototype = JsvServiceClient.prototype;
```
Comment (aallfredo) [inline ChromeExtension/js/GeniusLinkServiceClient.js]: So this is wrong all you need to do is add the method to the original object not create one new object for each method.

This should look like

`function GeniusLinkServiceClient(baseUrl, apiKey, apiSecret) {
 +    var that = this;
 +    JsvServiceClient.call(this, baseUrl);
 +    that.baseSyncReplyUri = baseUrl;
 +    that.baseAsyncOneWayUri = baseUrl;
 +    that.apiSecret = apiSecret;
 +    that.apiKey = apiKey;
 +    that.getFromService = function (url, objectToProcess, callback,error) {
 +        objectToProcess.apiKey = that.apiKey;
 +        objectToProcess.apiSecret = that.apiSecret;
 +        objectToProcess.format = "jsv";
 +
 +        GeniusLinkServiceClient.prototype.getFromService.call(this, url, objectToProcess, callback, error);
 +    };
 +
 + that.postFormDataToService = function (url, objectToProcess, callback, error) {
 +        objectToProcess.apiKey = that.apiKey;
 +        objectToProcess.apiSecret = that.apiSecret;
 +
 +        GeniusLinkServiceClient.prototype.postFormDataToService.call(this, url, objectToProcess,
 +            callback, error);
 +    };
 +
that.postToService = function (url, objectToProcess, callback, error) {
 +        objectToProcess.apiKey = that.apiKey;
 +        objectToProcess.apiSecret = that.apiSecret;
 +
 +        GeniusLinkServiceClient.prototype.postFormDataToService.call(this, url, objectToProcess,
 +            callback, error);
 +    };
 +
 +}`

File: ChromeExtension/js/apiKeyViewModel.js [hunk 2]
```diff
@@ -1,3 +1,7 @@
+function localStorageHasValue(val) {
```
Comment (aallfredo) [inline ChromeExtension/js/apiKeyViewModel.js]: Put this function on its own file and just refer to it

File: ChromeExtension/js/background.js [hunk 3]
```diff
@@ -8,6 +8,9 @@ function getCurrentTab() {
         });
     });
 }
+function localStorageHasValue(val) {
```
Comment (aallfredo) [inline ChromeExtension/js/background.js]: That way we are not declaring it twice.

File: ChromeExtension/js/prueba.js [hunk 4]
```diff
@@ -0,0 +1 @@
+alert("hola");
```
Comment (aallfredo) [inline ChromeExtension/js/prueba.js]: erase this

File: ChromeExtension/js/lastlinks.js [hunk 5]
```diff
@@ -31,12 +32,11 @@ document.addEventListener('DOMContentLoaded', function() {
         console.error("Error: ", error);
     });
 
-    $('#results').on('click', 'a', function() {
+    $('#results').on('click', 'a', function () {
```
Comment (aallfredo) [inline ChromeExtension/js/lastlinks.js]: This should be on knockout

File: ChromeExtension/js/background.js [hunk 6]
```diff
@@ -42,32 +45,27 @@ function createGeniusCurrentLink(e) {
...
-    var newLink;
-    req.open('POST', "https://api.geni.us/v2/shorturl?apiKey=" + localStorage["apiKey"] + "&apiSecret=" + localStorage["apiSecret"] + "&format=json&GroupId=" + localStorage["defaultGroupId"] + "&bulkMode=0&domain=geni.us&Url=" + url);
-    req.setRequestHeader("Content-type", "application/x-www-form-urlencoded");
-    req.onload = function () {
-        if (req.status == 200 && req.readyState === 4) {
-            newLink = JSON.parse(req.response)['NewLink'];
+
+ // no entiendo porque no coge los parametros que le envio en el segundo paramentro(spot) de la funcion
+    var client = new GeniusLinkServiceClientPost("https://api.geni.us/v2", localStorage["apiKey"], localStorage["apiSecret"]);
+    client.postToService("shorturl?apiKey=" + localStorage["apiKey"] + "&apiSecret=" + localStorage["apiSecret"] + "&Url=" + url + "&GroupId=" + localStorage["defaultGroupId"] + "&bulkMode=0",
```
Comment (aallfredo) [inline ChromeExtension/js/background.js]: This doesn't look correct.

Comment (aallfredo) [inline ChromeExtension/js/background.js]: We should be passing the parameters in the body no the url and the apiKey and The apiSecret should be added in the internal method.

Comment (aallfredo) [inline ChromeExtension/js/background.js]: I showed you that on skype in the afternoon.

**Comment by mmesson** [comment]
@aallfredo please review last commit

File: ChromeExtension/manifest.json [hunk 7]
```diff
@@ -16,7 +16,8 @@
     },
 
     "background": {
-        "scripts": ["js/globalfunctions.js", "js/background.js", "js/jquery.min.js"],
+        "scripts": ["js/servicestack.js" , "js/GeniusLinkServiceClient.js", "js/background.js", "js/jquery.min.js"],
+        // "page": "prueba.html",
```
Comment (aallfredo) [inline ChromeExtension/manifest.json]: erase this and then we can merge

**Pull request merged**